framework/src/main/java/com/github/dbmdz/flusswerk/framework/engine/DefaultEngine.java

```diff
@@ -142,6 +152,7 @@ public void process(Message message) {
           new StopProcessingException("Could not finish message handling").causedBy(e);
       fail(message, stopProcessingException);
     }
+    tracing.deregister();
```

Comment:
müßte das nicht eher in einen Finally-Block gehen?

Reply:
Stimmt, hab's ergänzt.